### PR TITLE
[SPARK-42316][SQL] Assign name to _LEGACY_ERROR_TEMP_2044

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
This PR proposes to assign name to _LEGACY_ERROR_TEMP_2044, "BINARY_ARITHMETIC_OVERFLOW".

### Why are the changes needed?
Assign proper name to LEGACY_ERROR_TEMP

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
 ./build/sbt "testOnly org.apache.spark.sql.errors.QueryExecutionErrorsSuite"

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 1]
```diff
@@ -625,6 +625,20 @@ class QueryExecutionErrorsSuite
...
 
+  test("BINARY_ARITHMETIC_CAUSE_OVERFLOW: byte plus byte result overflow") {
+    checkError(
+      exception = intercept[SparkArithmeticException] {
+        sql(s"select CAST('5' AS TINYINT) + CAST('5' AS TINYINT)").collect()
+      },
+      errorClass = "BINARY_ARITHMETIC_CAUSE_OVERFLOW",
+      parameters = Map(
+        "value1" -> "5",
+        "symbol" -> "+",
```
[the remark this answers is not in the document]
Comment (Hisoka-X) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:636]: Hi, I use `select CAST('32767' AS TINYINT) + CAST('32767' AS TINYINT)` but result null, is it acceptable？ If not, I will fix it.
image

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 2]
```diff
@@ -625,6 +625,20 @@ class QueryExecutionErrorsSuite
...
+        "value2" -> "5",
+      ),
```
[the remark this answers is not in the document]
Comment (Hisoka-X) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala]: Thanks for metion it! I can't find this scala style problem in local, the sbt build report error is not clear enough.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hisoka-X) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala]: > Hi, I use `select CAST('32767' AS TINYINT) + CAST('32767' AS TINYINT)` but result null, is it acceptable？ If not, I will fix it. image

Forget it, I know why. Thanks.

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -35,6 +35,12 @@
     ],
     "sqlState" : "22003"
   },
+  "BINARY_ARITHMETIC_CAUSE_OVERFLOW" : {
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:38]: Let's make it shorter `BINARY_ARITHMETIC_OVERFLOW` similarly to other error class names in the file.

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 4]
```diff
@@ -625,6 +625,21 @@ class QueryExecutionErrorsSuite
     }
   }
 
+  test("BINARY_ARITHMETIC_CAUSE_OVERFLOW: byte plus byte result overflow") {
+    withSQLConf("spark.sql.ansi.enabled" -> "true") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:629]:
```suggestion
    withSQLConf(SQLConf.ANSI_ENABLED.key -> "true") {
```

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala [hunk 5]
```diff
@@ -625,6 +625,21 @@ class QueryExecutionErrorsSuite
...
+      checkError(
+        exception = intercept[SparkArithmeticException] {
+          sql(s"select CAST('127' AS TINYINT) + CAST('5' AS TINYINT)").collect()
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala:632]: You can create a TINYINT constant directly without casting:
```suggestion
          sql(s"select 127Y + 5Y").collect()
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala (hunk 4 above)
Comment (Hisoka-X) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala]: Done

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala (hunk 5 above)
Comment (Hisoka-X) [reply sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala]: Thanks for your reminder. Done

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
@Hisoka-X Please, update PR's description and fix the error class name.

Is `fanjia` at https://issues.apache.org/jira/browse/SPARK-42316 your JIRA account ?

**Comment by Hisoka-X** [comment]
> @Hisoka-X Please, update PR's description and fix the error class name.

Done

> Is `fanjia` at https://issues.apache.org/jira/browse/SPARK-42316 your JIRA account ?

Yes

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @Hisoka-X.

**Comment by Hisoka-X** [comment]
Thank @MaxGekk

**Pull request closed without merging**